pkg/policy/api/ingress.go

```diff
@@ -29,13 +29,6 @@ import (
 //   the rule to take effect. The exception to this rule is FromRequires field;
 //   the effects of any Requires field in any rule will apply to all other
 //   rules as well.
-//
-// - For now, combining ToPorts, FromCIDR, and FromEndpoints in the same rule
-//   is not supported and any such rules will be rejected. In the future, this
-//   will be supported and if multiple members of this structure are specified,
-//   then all members must match in order for the rule to take effect. The
-//   exception to this rule is the Requires field, the effects of any Requires
-//   field in any rule will apply to all other rules as well.
```

Comment:
We can probably simplify this to:

`Combining FromCIDR and FromEndpoints in the same rule is not allowed.`

Reply:
Sure. Where is that enforced?

Reply:
It should be done by `sanitize()` in `pkg/policy/api/rule_validation.go`.

Note that the policy above is two rules, one with `fromEndpoints` and the other with `toPorts` + `fromCIDR`. This is valid. The `-` before `toPorts` would need to be removed to test this.